src/internal/connector/onedrive/restore.go

```diff
@@ -211,7 +219,7 @@ func RestoreCollection(
 				continue
 			}
 
-			if source == OneDriveSource {
+			if source == OneDriveSource && backupVersion == versionWithDataAndMetaFiles {
```

Comment:
We might have to update the checks in future to either a range check or do a greater than check. Keeping it this way for now as. The tests would fail when someone bumps us the version which should give a hint about this and I'm thinking we could consider more about it at  that point.